### PR TITLE
Create test module: Vsftpd

### PR DESCRIPTION
**Pull request by ShawnHa0**
Testing vsftpd and lftp with ssl enabled. There are bugs related to this test module currently. Thus further code enhancement is anticipated when bugs are fixed or discovered acceptable workarounds. Test verification is expected to fail due to those bugs. And testing mechanisms can be enhanced once bugs are fixed.

- Related ticket: https://progress.opensuse.org/issues/108614
- Needles: N/A
- Verification run: https://openqa.opensuse.org/tests/2350748#details

### CODE REVIEW

File: tests/security/vsftpd/lftp.pm [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    enter_cmd('cd files/');
+    enter_cmd('ls');
+    enter_cmd('get f1.txt');
+    enter_cmd('exit');
+
+    # Check if file has been uploaded
+    assert_script_run('ls | grep f1.txt');
+
+    # Upload a file to directory
+    assert_script_run('touch f2.txt');
```
Comment (rfan1) [inline tests/security/vsftpd/lftp.pm:39]: I suggest you add some context to f1.txt and f2.txt, 
and if possible, md5 check for both file should be better

File: tests/security/vsftpd/vsftpd_setup.pm [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    assert_script_run("mkdir -p $user_path");
+    assert_script_run("useradd -d $user_path $user");
+    assert_script_run("echo $user:$pwd | chpasswd");
+
+    # Grant root ownership, create a separate directory for uploading and give it to ftpuser
+    assert_script_run("chown root:root $user_path");
+    assert_script_run("mkdir -p $ftp_file_path");
+    assert_script_run("chown $user:users $ftp_file_path");
+
+    # New user needs access to ttyS0
```
Comment (rfan1) [inline tests/security/vsftpd/vsftpd_setup.pm:67]: Good change, l suggest you add more comment for the default permission 620, then we need change it

File: tests/security/vsftpd/vsftpd.pm [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+use testapi;
+use utils;
+
+sub run {
+    my $ftp_file_path = '/srv/ftp/ftpuser/files';
+    my $user = 'ftpuser';
+
+    select_console 'root-console';
+
+    # Create a file
```
Comment (rfan1) [inline tests/security/vsftpd/vsftpd.pm:20]: Add more comments here?

File: tests/security/vsftpd/lftp.pm [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    enter_cmd("lftp -d -u $user,$pwd -e 'set ftp:ssl-force true' localhost");
+    enter_cmd('ls');
+    enter_cmd('cd files/');
+    enter_cmd('ls');
+    enter_cmd('put f2.txt');
+    enter_cmd('exit');
```
Comment (Amrysliu) [inline tests/security/vsftpd/lftp.pm:45]: These codes could be written in a `for` loop to avoid repeating?

**Review by jouyingbin — APPROVED** [review]
LGTM, and we will enhance the code in the future too.
Thank you.

**Pull request merged**